openquakeplatform/bin/oq_install.sh

```diff
@@ -0,0 +1,227 @@
+    cp -rp $INSTALL_DIR/support/geonode.robots $GEONODE_WWW/robots.txt
+    cp -rp $INSTALL_DIR/support/geonode.binary $GEONODE_BIN/geonode
+    cp -rp $INSTALL_DIR/GeoNode*.zip $GEONODE_SHARE
+    cp -rp $INSTALL_DIR/support/geonode.updateip $GEONODE_BIN/geonode-updateip
+    cp -rp $INSTALL_DIR/support/geonode.admin $GEONODE_SHARE/admin.json
+    cp -rp $INSTALL_DIR/support/geonode.local_settings $GEONODE_ETC/local_settings.py
+
+    chmod +x $GEONODE_BIN/geonode
+    chmod +x $GEONODE_BIN/geonode-updateip
+}
+
+function preinstall() {
+    setup_directories
+    reorganize_configuration
+    echo "Fine preinstall"
+}
+
+function randpass() {
+    [ "$2" == "0" ] && CHAR="[:alnum:]" || CHAR="[:graph:]"
+    cat /dev/urandom | tr -cd "$CHAR" | head -c ${1:-32}
+    echo
+}
+
+function setup_postgres_every_time() {
+    true # nothing to do. when we setup DB migrations they'll probably go here.
+}
+
+function setup_django_every_time() {
+    echo "setup_django_every_time:"
+    source ~/env/bin/activate
+    
+    pip -v install /usr/share/geonode/GeoNode-*.zip --no-dependencies --quiet
+    geonodedir=`python -c "import geonode;import os;print os.path.dirname(geonode.__file__)"`
+
+    ln -sf /etc/geonode/local_settings.py $HOME/env/lib/python2.7/site-packages/geonode/local_settings.py
+    ln -sf /usr/lib/python2.7/dist-packages/osgeo $HOME/env/lib/python2.7/site-packages
+
+    # Set up logging symlink
+    mkdir -p $GEONODE_LOG
+    ln -sf /var/log/apache2/error.log $GEONODE_LOG/apache.log
+
+    export DJANGO_SETTINGS_MODULE=geonode.settings
+
+    django-admin migrate account --settings=geonode.settings
+    # geonode runserver &
+    geonode migrate --verbosity 0
+    geonode loaddata $geonodedir/base/fixtures/initial_data.json
+    geonode collectstatic --noinput --verbosity 0
+    geonode createsuperuser
+
+    # Create an empty uploads dir
+    mkdir -p $GEONODE_WWW/uploaded
+    # mkdir -p $GEONODE_WWW/uploaded/thumbs/
+    ln -sf $HOME/env/lib/python2.7/site-packages/geonode/uploaded/thumbs $GEONODE_WWW/uploaded
+    mkdir -p $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded
+    mkdir -p $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded/layers
+    mkdir -p $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded/documents
+
+    # Apply the permissions to the newly created folders.
+    sudo chown www-data -R $GEONODE_WWW
+
+    # Open up the permissions of the media folders so the python
+    # processes like updatelayers and collectstatic can write here
+    chmod 777 -R $GEONODE_WWW/uploaded
+    chmod 777 -R $GEONODE_WWW/static
+    chmod 777 -R $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded/
+    chmod 777 -R $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded/layers
+    chmod 777 -R $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded/documents
+    chmod 777 -R $HOME/env/local/lib/python2.7/site-packages/geonode/static_root
```

Comment:
As said above, never use `777` it is forbidden. Please fix permissions and ownership to allow max `775` for folders and `664` (or less) for files.

Using `chmod -R` applies the permission to all files and folders without distinction. It should be used only if you are sure that the tree contains only folders.

`chmod 777 -R $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded/layers` and `chmod 777 -R $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded/documents` are anyway redundant since are covered by `chmod 777 -R $HOME/env/local/lib/python2.7/site-packages/geonode/uploaded/`.

I would suggest you to review entirely the management of static data and uploaded data. Their path can be customized via `STATIC_URL` and `MEDIA_URL` via `local_settings.py`.
If you need to apply permission recursively use `find`:
- `find . -type f -exec chmod 644 '{}' \;` for files
- `find . -type d -exec chmod 755 '{}' \;` for folders

Reply:
This script is in Geonode package, i have just putted other thinks for the installation

Reply:
Ok, but I will not let `777` folders to be installed on a production server, so it must be fixed (since we are already using a copy of it, it should be easy)